main/EDInitMod.F90

```diff
@@ -293,11 +301,40 @@ subroutine set_site_properties( nsites, sites,bc_in )
           sites(s)%NF         = 0.0_r8         
           sites(s)%frac_burnt = 0.0_r8
          
-         ! PLACEHOLDER FOR PFT AREA DATA MOVED ACROSS INTERFACE                                                                                   
-          if(hlm_use_fixed_biogeog.eq.itrue)then
+
+          if(hlm_use_fixed_biogeog.eq.itrue)then  
+           ! MAPPING OF FATES PFTs on to HLM_PFTs
+            ! add up the area associated with each FATES PFT
+            ! where pft_areafrac is the area of land in each HLM PFT and (from surface dataset)
+            ! hlm_pft_map is the area of that land in each FATES PFT (from param file)
+            sites(s)%area_pft(1:numpft) = 0._r8
+            do hlm_pft = 1,size( EDPftvarcon_inst%hlm_pft_map,2)
+               do fates_pft = 1,numpft ! loop round all fates pfts for all hlm pfts        
+                 sites(s)%area_pft(fates_pft) = sites(s)%area_pft(fates_pft) + &
+                     EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft) * bc_in(s)%pft_areafrac(hlm_pft) 
+               end do
+            end do !hlm_pft
+
+           ! re-normalize PFT area to ensure it sums to one.
```

Comment:
is this to handle precision errors or to handle cases where the above (rows and columns don't sum to one) isn't satisfied?  if the latter, how do we know that the numbers make physical sense?

Reply:
It's really to handle bare ground.  i.e. in cases where area_pft(1:numpft) is less than one. As I have it now, this bare ground gets subsumed into the PFT areas. This made sense when I was thinking about -only- nocomp mode, (as I had thought that the unclosed canopy in FATES, where things don't grow, should in principle generate bare ground  but actually in SP mode, it's going to cause all of the areas where there is lots of bare ground to have the LAI of the other plant functional types and therefore a big positive discrepancy in grid cell LAI between the surface dataset and FATES. So i need to think of a better way of handling bare ground in SP mode... 

Also it might well be true in dynamic nocomp mode that we want to specific an area of bare ground. I should extend that logic as well.